tests/test_routing.py

```diff
@@ -417,6 +417,14 @@ def test_reverse_mount_urls() -> None:
         mounted.url_path_for("users", subpath="test", path="/tom") == "/test/users/tom"
     )
 
+    mounted = Router([Mount("/users", ok, name="users")])
+    with pytest.raises(NoMatchFound):
+        mounted.url_path_for("users", path="/a", foo="bar")
+
+    mounted = Router([Mount("/users", ok, name="users")])
+    with pytest.raises(NoMatchFound):
+        mounted.url_path_for("users")
```

Comment:
Testing the cases where the if conditions in these two logic blocks are not true.
https://github.com/encode/starlette/blob/master/starlette/routing.py#L456
https://github.com/encode/starlette/blob/master/starlette/routing.py#L458